api/cas/views.py

```diff
@@ -0,0 +1,310 @@
+
+
+class LoginExternal(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
+    """ Login through non-institution external IdP. Authentication Required.
+    """
+
+    view_category = 'cas'
+    view_name = 'login-external'
+    permission_classes = (permissions.IsCasLogin,)
+    authentication_classes = (login.CasLoginAuthentication,)
+
+    def post(self, request, *args, **kwargs):
+
+        content = {
+            'username': request.user.username
+        }
+
+        return Response(data=content, status=status.HTTP_200_OK)
+
+
+class AccountRegisterOsf(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
+    """ Default account creation through OSF.
+    """
+
+    view_category = 'cas'
+    view_name = 'account-register-osf'
+    permission_classes = ()
+    authentication_classes = ()
+
+    def post(self, request, *args, **kwargs):
+
+        self.load_request_body_data(request)
+        body_data = json.loads(request.body['data'])
+        account_action = body_data.get('accountAction')
+
+        if account_action != 'REGISTER_OSF':
+            raise ValidationError(detail=messages.INVALID_REQUEST)
+
+        account.handle_register_osf(body_data.get('user'))
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class AccountVerifyOsf(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
+    """ Verify email for account creation through OSF.
+    """
+
+    view_category = 'cas'
+    view_name = 'account-verify-osf'
+    permission_classes = ()
+    authentication_classes = ()
+
+    def post(self, request, *args, **kwargs):
+
+        body_data = self.load_request_body_data(request)
+        account_action = body_data.get('accountAction')
+
+        if account_action != 'VERIFY_OSF':
+            raise ValidationError(detail=messages.INVALID_REQUEST)
+
+        user = account.handle_verify_osf(body_data.get('user'))
+
+        content = {
+            'verificationKey': user.verification_key,
+            'userId': user._id,
+            'casAction': 'account-verify-osf',
+            'nextUrl': False,
+        }
+
+        return Response(data=content, status=status.HTTP_200_OK)
+
+
+class AccountVerifyOsfResend(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
+    """ Find user by email and resend verification email for account creation.
+    """
+
+    view_category = 'cas'
+    view_name = 'account-verify-osf-resend'
+    permission_classes = ()
+    authentication_classes = ()
+
+    def post(self, request, *args, **kwargs):
+
+        body_data = self.load_request_body_data(request)
+        account_action = body_data.get('accountAction')
+
+        if account_action != 'VERIFY_OSF_RESEND':
+            raise ValidationError(detail=messages.INVALID_REQUEST)
+
+        account.handle_verify_osf_resend(body_data.get('user'))
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class AccountRegisterExternal(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
```

Comment:
Create or link OSF account for user authenticated through external identity provider (ORCiD only for now).